### PR TITLE
#161779875 create route to list all users

### PR DESCRIPTION
**Pull request by Danielshow**
#### What does this PR do?
This PR create an endpoint for listing authors profile 

#### Description of Task to be completed?
- create route to list users
- add test
- fix bug in user signup test

#### How should this be manually tested?
- clone this branch `ft-list-users-functionality-#161779875`
- run `npm install`
- post `/api/v1/users`
- run test `npm test`

#### Any background context you want to provide?
- Only an authenticated user can view others profile, Login to get your token

#### What are the relevant pivotal tracker stories?
[#161779875](https://www.pivotaltracker.com/story/show/161779875)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: server/helpers/removeDateStampAndPassword.js [hunk 1]
```diff
@@ -5,12 +5,12 @@
...
-   if(!user.firstName){
-     throw new Error('Please enter a valid user object');
-   }
-   const {createdAt, updatedAt, password, ...userObject } = user;
-   return userObject;
+const removeDateStampsAndPassword = (user) => {
+  if(!user.firstName){
+    throw new Error('Please enter a valid user object');
+  }
+  const {createdAt, updatedAt, password, ...userObject } = user;
```
Comment (theghostyced) [inline server/helpers/removeDateStampAndPassword.js:12]: Love that you used a spread operator here. Good job.

**Review by theghostyced — APPROVED** [review]
LGTM @Danielshow @efalayi

File: server/controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+const { User, Profile } = models;
+/**
+ * @description - Performs all auth function
+ */
+class UserController {
+  /**
+   * @param {object} req - req sent to server
+   * @param {object} res - response gotten from server
+   * @param {fn} next - csllback function
```
Comment (efalayi) [inline server/controllers/userController.js]: A typo here: `csllback`should be `callback`

File: tests/users/signIn.test.js [hunk 3]
```diff
@@ -126,4 +128,22 @@ describe('API endpoint for POST auth/signin - Email Validations', () => {
...
+      password: 'danielshow2#'
+    };
+    const req={
+      body: user
+    };
+
+    const res = {};
+
+    const next = sinon.stub();
+    sinon.stub(User, 'findOne').throws();
```
Comment (efalayi) [inline tests/users/signIn.test.js:144]: Good job 👍

File: server/controllers/userController.js (hunk 2 above)
Comment (Danielshow) [reply server/controllers/userController.js]: Done, thanks.

**Review by efalayi — APPROVED** [review]
LGTM

**Pull request merged**